### DISCUSSION TITLE
Additional properties in alert contributing events?

### DISCUSSION THREAD

**Discussion by chuseman** [datalust/seq-tickets#1866] [Help]
Hello, I'm wondering if it is possible to include additional properties in the Contributing Events for an alert?

Right now, I only see the event ID, timestamp, and message in the `Source.ContributingEvents` array as described in the docs.

Ideally, I'd like to include the exception and correlation ID properties.

Is this possible?

**Comment by nblumhardt** [comment]
Hi Chris,

It's not possible today, though it's something we'll give some thought to.

If you can live without batching of events into a single notification, you can use "Stream incoming events" on a Seq App instance to receive errors (etc.) one at a time, with full details, as they arrive at the server. Let me know if you need any more info on this option.

Hope this helps,
Nick

**Reply by chuseman** [reply]
Nick, any progress on this front at all?

I don't think "Stream incoming events" will work for our use case (we still need the aggregation from alerts). I'd like to push for more Seq adoption in my company but it's a tough sell without this.

Is the only option to have this today to write my own Seq App that can turn around and query Seq for each event ID when it receives the alert?

**Reply by nblumhardt** [reply]
Thanks for the feedback, Chris.

I'll create a ticket from this discussion; I don't think it'll be too hard for us to make a change that makes more properties available. As things are today, writing your own app to do it is probably the only option, unfortunately.

**Reply by nblumhardt** [reply]
https://github.com/datalust/seq-tickets/issues/2364

**Reply by chuseman** [reply]
Any movement on this issue?

**Reply by nblumhardt** [reply]
Hi Chris! We're currently in the deeper parts of metrics support for Seq 2026.1; our plan is to get a public preview of that out (ASAP), and then during the release stabilization we'll work through some of the more self-contained items from the tracker. We'll update #2364 as soon as we have an update 👍

**Reply by Edgaras91** [reply]
Hi @nblumhardt,

I just came across the “Stream incoming events” feature from your comment, and it’s exactly what I’ve been trying to achieve with alerts. Ideally, we’d also need aggregation, while including all properties and surfacing the error message in the subject (presumably from the first event).
Do you think something like this would be possible?

**Reply by nblumhardt** [reply]
Hi @Edgaras91! What you're describing is essentially the extension to "include contributing events" that attaches additional properties. We're in the process of pushing the first reasonably complete 2026.1 metrics previews now, so we're just about ready to loop back around to this :-)

**Comment by c0shea** [comment]
This would be a great feature to be able to loop over all the contributing events and pretty print their properties in the email. I love the newer Alerts feature/dashboard, but it doesn't output as nice of emails as the legacy stream incoming events path. But the legacy multiple app instances path is hard to manage with so many configured instances.

**Comment by schallm** [comment]
Our developers are asking for this as well.  It is great to only alert base on the velocity of errors occurring, but many times the decision point of who needs to handle the issue is based on the exception text of the contributing events.  In most cases in our environments, the Seq server is not available on the internet, so the decision on who call/assign the issue to can only be done after accessing the environment rather than from the alert email.
